src/content/advanced/vsphere-kubevip/index.md

```diff
@@ -0,0 +1,91 @@
+  - What IP should I use for Kube-vip with CAPV?
+  - How to set a range of IPs for load balancer in Kube-vip with CAPV?
+  - Where can I see the IPs in use by Kube-vip?
+aliases:
+  - /guides/kube-vip-for-capv/
+owner:
+  - https://github.com/orgs/giantswarm/teams/team-rocket
+---
+While most infrastructure providers offer a load balancer to use for the Kubernetes API and services of type Load Balancer (AWS, Azure, VMware Cloud Director...), vSphere is a bit different in that it is a hypervisor as opposed to a cloud provider. Advanced networking use cases such as routing, firewalling, load balancing and the likes are handled by VMware NSX. In order to offer multi-control plane nodes to all vSphere customers (even those without NSX), CAPV comes with `Kube-vip`, a layer-2 load balancer.
+
+As a result of working on layer 2, `Kube-vip` uses IP addresses in the same subnet as the nodes themselves and these IP addresses must be selected carefully to avoid collisions. This is particularly plausible if multiple operators collaborate on the platform.
+
+## IPAM controller
+
+in order to mitigate this risk, Giant Swarm CAPV clusters run `cluster-api-ipam-provider-in-cluster` to track IP addresses usage from within the management cluster (MC). For that reason, **we recommend running a maximum of one management cluster per subnet**.
+
+The basic operations of `cluster-api-ipam-provider-in-cluster` work as follows:
+
+* Create a pool of IPs with a `globalinclusterippools` custom resource (CR).
+* Claim an IP by referencing the pool in a `ipaddressclaims` CR.
+* An `ipaddresses` CR is created and the `Free/Used` status of the `globalinclusterippools` CR is updated.
+
+```sh
+> k get globalinclusterippools.ipam.cluster.x-k8s.io
+NAME        ADDRESSES                         TOTAL   FREE   USED
+wc-cp-ips   ["10.10.222.232-10.10.222.239"]   8       7      1
+
+> k get ipaddressclaims.ipam.cluster.x-k8s.io -A
+NAMESPACE        NAME    POOL NAME   POOL KIND
+org-giantswarm   pssss   wc-cp-ips   GlobalInClusterIPPool
+
+NAMESPACE        NAME    ADDRESS         POOL NAME   POOL KIND
+org-giantswarm   pssss   10.10.222.233   wc-cp-ips   GlobalInClusterIPPool
+```
+
+You can find more details about how it works in this [blog post](https://kremser.dev/post/ipam-for-capv/).
+
+## Slicing the subnet
+
+Here we will talk about how to "slice" the subnet where you will be deploying your workload clusters (provided they will all be on the same subnet of course).
+
+The main goal here is to avoid IP address collisions so we want the subnet to accomodate for:
+
+* DHCP range for node interface.
+* One IP for the management clusters' Kubernetes API.
+* One IP for the management clusters' Ingress.
+* Pool of IPs for workload clusters' Kubernetes API.
+* Pool of IPs for workload clusters' Ingresses (untracked by IPAM controller).
+
+Below is an example of what it could look like but you are free to adjust the ranges as you see fit:
+
+![kube-vip range](capv-kubevip-ipam.png)
```

Comment:
We normally provide the editable image source so others can edit the diagram later.

For example:

```
<!-- Source for above image: https://docs.google.com/drawings/d/1PDve3HoE7br_6npe0RSCw8ddt-H7pFztPlSDiQbNabs/edit -->
```

Or alternatively, if the diagram is from draw.io, export a PNG including the source.

Reply:
It's from excalidraw (like draw.io).
TIL you can embed the source in the png